src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts

```diff
@@ -666,6 +655,37 @@ export class DatasetFormComponent implements FormConfiguration {
           ],
           required: false,
           value: DatasetAclType.Inherit,
+          onChangeOption: (event: { event: MatSelectChange }) => {
+            const aclModeFormControl = this.entityForm.formGroup.get('aclmode') as FormControl;
+            const value = event.event.value;
+            if (value === DatasetAclType.Nfsv4) {
+              aclModeFormControl.setValue(AclMode.Passthrough);
+              this.entityForm.setDisabled('aclmode', false, false);
+            } else if (value === DatasetAclType.Posix) {
+              aclModeFormControl.setValue(AclMode.Discard);
+              this.entityForm.setDisabled('aclmode', true, false);
+            } else if (value === DatasetAclType.Inherit) {
+              aclModeFormControl.setValue(AclMode.Inherit);
+              this.entityForm.setDisabled('aclmode', true, false);
```

Comment:
When add form is opened, acltype is `inherit` by default, but `aclmode` is not disabled.